wallet/common/symbol/src/utils/transaction-from-dto.js

```diff
@@ -0,0 +1,635 @@
+import { addressFromPublicKey, addressFromRaw } from './account';
+import { networkTimestampToUnix } from './helper';
+import {
+	formatMosaicList,
+	getMosaicAmount,
+	isRestrictableFlag,
+	isRevokableFlag,
+	isSupplyMutableFlag,
+	isTransferableFlag
+} from './mosaic';
+import { networkTypeToIdentifier } from './network';
+import {
+	createFee,
+	decodePlainMessage,
+	getUnresolvedIdsFromTransactions,
+	isAggregateTransaction,
+	isIncomingTransaction,
+	isOutgoingTransaction
+} from './transaction';
+import {
+	AddressRestrictionFlagMessage,
+	AliasActionMessage,
+	LinkActionMessage,
+	LockHashAlgorithmMessage,
+	Message,
+	MessageType,
+	MosaicRestrictionFlagMessage,
+	MosaicRestrictionTypeMessage,
+	MosaicSupplyChangeActionMessage,
+	NamespaceRegistrationTypeMessage,
+	OperationRestrictionFlagMessage,
+	TransactionType
+} from '../constants';
+import { SdkError, absoluteToRelativeAmount, safeOperationWithRelativeAmounts } from 'wallet-common-core';
+
+/** @typedef {import('../types/Account').PublicAccount} PublicAccount */
+/** @typedef {import('../types/Mosaic').Mosaic} Mosaic */
+/** @typedef {import('../types/Network').NetworkProperties} NetworkProperties */
+/** @typedef {import('../types/Transaction').Transaction} Transaction */
+
+export const isAggregateTransactionDTO = transactionDTO => {
+	const { transaction } = transactionDTO;
+
+	return isAggregateTransaction(transaction);
+};
+
+const createMosaic = (id, amount) => mosaicFromDTO({ id, amount });
+
+const mosaicFromDTO = mosaic => ({
+	id: mosaic.id,
+	amount: mosaic.amount
+});
+
+const formatAddress = (rawAddress, resolvedAddresses) => {
+	if (rawAddress.length === 48)
+		return addressFromRaw(rawAddress);
+
+	return resolvedAddresses[rawAddress] || null;
+};
+
+/**
+ * Converts a transaction DTO to a transaction object.
+ * @param {object} transactionDTO - The transaction to convert.
+ * @param {object} config - The configuration object.
+ * @param {NetworkProperties} config.networkProperties - The network properties.
```

Comment:
are the other param after config needed?

Reply:
Yes, these parameters are needed. They’re used to correctly map and resolve transactions.